src/subdataframe/subdataframe.jl

```diff
@@ -112,12 +112,21 @@ index(sdf::SubDataFrame) = index(sdf.parent)
 nrow(sdf::SubDataFrame) = ncol(sdf) > 0 ? length(sdf.rows)::Int : 0
 ncol(sdf::SubDataFrame) = length(index(sdf))
 
+function Base.getindex(sdf::SubDataFrame, colind::Union{ColumnIndex, Colon})
+    return sdf.parent[sdf.rows, colind]
```

Comment:
What's the logic for keeping the current behavior with the `Colon`? For the single columns of `ColumnIndex` a Vector/DataVector will be returned, so it makes sense to me there. But for the `Colon` case, that will return a `DataFrame` again rather than a `SubDataFrame`. Is that your desired behavior here @joshbode? 

```julia
julia> subdf
10×3 DataFrames.SubDataFrame{Array{Int64,1}}
│ Row │ x        │ y         │ z        │
├─────┼──────────┼───────────┼──────────┤
│ 1   │ 0.845989 │ 0.527821  │ 0.270826 │
│ 2   │ 0.816581 │ 0.612666  │ 0.609996 │
│ 3   │ 0.893455 │ 0.608707  │ 0.325525 │
│ 4   │ 0.887801 │ 0.0797759 │ 0.275525 │
│ 5   │ 0.266899 │ 0.675255  │ 0.577029 │
│ 6   │ 0.837197 │ 0.107275  │ 0.591336 │
│ 7   │ 0.347261 │ 0.489731  │ 0.785748 │
│ 8   │ 0.245108 │ 0.106559  │ 0.961957 │
│ 9   │ 0.529789 │ 0.0781499 │ 0.839842 │
│ 10  │ 0.75061  │ 0.562225  │ 0.532867 │

julia> subdf.parent[subdf.rows, 1]
10-element DataArrays.DataArray{Float64,1}:
 0.845989
 0.816581
 0.893455
 0.887801
 0.266899
 0.837197
 0.347261
 0.245108
 0.529789
 0.75061

julia> subdf.parent[subdf.rows, :]
10×3 DataFrames.DataFrame
│ Row │ x        │ y         │ z        │
├─────┼──────────┼───────────┼──────────┤
│ 1   │ 0.845989 │ 0.527821  │ 0.270826 │
│ 2   │ 0.816581 │ 0.612666  │ 0.609996 │
│ 3   │ 0.893455 │ 0.608707  │ 0.325525 │
│ 4   │ 0.887801 │ 0.0797759 │ 0.275525 │
│ 5   │ 0.266899 │ 0.675255  │ 0.577029 │
│ 6   │ 0.837197 │ 0.107275  │ 0.591336 │
│ 7   │ 0.347261 │ 0.489731  │ 0.785748 │
│ 8   │ 0.245108 │ 0.106559  │ 0.961957 │
│ 9   │ 0.529789 │ 0.0781499 │ 0.839842 │
│ 10  │ 0.75061  │ 0.562225  │ 0.532867 │
```